### PR TITLE
feat: DTMF keypad support for the kit and Console template

### PR DESCRIPTION
**Pull request by Regaddi**
## Summary

Adds DTMF (touch-tone) keypad support to the kit and the Console template.

- **New `DTMFKeypad` element** (`package/src/components/elements/DTMFKeypad.tsx`): a headless `DTMFKeypadComponent` plus a connected `DTMFKeypad` that sends tones via client-react's `useDTMF()` and stays disabled until the transport is connected. Two modes via a `mode` prop: `"immediate"` (default; each press sends its tone) and `"buffered"` (digits accumulate in an editable input with backspace + Send, dispatched as one `sendDTMF` sequence). Controlled/uncontrolled buffer via `value`/`defaultValue`/`onValueChange`; sends are wrapped in try/catch and surfaced through `onToneSent`/`onError`.
- **Console template**: the keypad lives behind a phone icon in the header control cluster — a Popover on desktop (sm+) and a bottom Drawer on mobile. Gated by a `noDTMF` prop; mode selectable via `keypadMode` (default `"immediate"`).
- **New `Drawer` UI primitive** (vaul) that portals into the scoped Pipecat UI container, mirroring the existing Popover pattern.
- Re-exports `useDTMF` from the kit hooks; exports `DTMFKeypad` from elements and `Drawer*` from ui. Stories cover immediate + buffered, pure + connected.

## Dependency changes

- Peer/dev floors bumped to `@pipecat-ai/client-js >=1.13.0` (sendDTMF/DTMFButton) and `@pipecat-ai/client-react >=1.8.0` (useDTMF), across root, `package/`, and `examples/01-console`. Verified empirically these are the exact minimums where those APIs first appear.
- Added `vaul ^1.1.2` to `package` deps for the Drawer (consistent with how the other Radix primitives are declared).

## Review fixes (this branch)

- `fix(dtmf)`: `buttonProps` was spread after the internal `onClick`/`disabled`, so a consumer's `buttonProps` could silently clobber the press handler or defeat the connected-state gate — now composed. Enter in buffered mode calls `preventDefault` so embedding in a `<form>` doesn't double-trigger a native submit.
- `fix(stories)`: the client-react `>=1.8.0` bump pulls in its broken published types (a duplicated, non-assignable `PipecatClient`), which regressed `tsc` on the new DTMF story and four pre-existing stories. Applied the same `as unknown as` cast already used in `PipecatAppBase`/`SessionInfo` to those call sites.

## Known workarounds (intentional — do not "clean up")

client-react@1.8.0 ships a duplicated `PipecatClient` class in its bundled `.d.ts` instead of re-exporting client-js's, so a client-js `PipecatClient` isn't assignable to `PipecatClientProvider`'s `client` prop. Two casts are intentional until a fixed client-react ships:
- `PipecatAppBase.tsx` — `as unknown as` cast at the provider boundary.
- `SessionInfo.tsx` — a now-unnecessary `@ts-expect-error` replaced with an explicit typed cast.
- (plus the five story casts added in this branch)

There is an upstream fix in `pipecat-ai/pipecat-client-web` (branch `fix/client-react-external-client-js-types`) that removes the path alias causing Parcel to inline client-js types. Once a client-react release with that fix ships, all of the above casts can be reverted and the client-react floor bumped.

## Follow-up (not blocking)

`Console/KeypadToggle` mounts an independent desktop Popover and mobile Drawer, toggled by the `sm:` breakpoint via CSS. If the viewport crosses the breakpoint while the keypad is open, the two overlays' open-state (and, in buffered mode, their separate buffers) aren't synchronized. Edge case (the Console default is `immediate`); a focused follow-up should render a single instance chosen by a media query.

## Test plan

- [x] `pnpm --filter=@pipecat-ai/voice-ui-kit build` (tsup: JS + DTS) passes
- [x] `pnpm --filter=@pipecat-ai/voice-ui-kit lint` passes
- [x] `pnpm --filter=@pipecat-ai/voice-ui-kit test` passes (7/7)
- [x] `pnpm install --frozen-lockfile` passes (pinned pnpm@10.14.0)
- [x] `tsc --noEmit` reports no DTMF/client-react errors (only pre-existing, unrelated errors remain)
- [ ] Manual: exercise the DTMF keypad stories (immediate + buffered, pure + connected) in Ladle

### CODE REVIEW

File: package/src/components/elements/DTMFKeypad.tsx [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+      if (buffered) setBuffer(buffer + key);
+    },
+    [buffered, buffer, onPress, setBuffer],
+  );
+
+  const handleSend = useCallback(() => {
+    if (!buffer) return;
+    onSend?.(buffer);
+    setBuffer("");
+  }, [buffer, onSend, setBuffer]);
```
[the remark this answers is not in the document]
Comment (Regaddi) [reply package/src/components/elements/DTMFKeypad.tsx:281]: Fixed in 4332b10. `handleSend` now runs `sanitizeDTMF` on the buffer before emitting, so a controlled `value`/`defaultValue` (which bypasses the input's onChange sanitization) can no longer produce an invalid DTMF sequence.

File: package/src/components/elements/DTMFKeypad.tsx [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+  const grid = (
+    <div className={cn("grid grid-cols-3 gap-2", classNames.container)}>
+      {DTMF_KEYS.map(({ value: key, sub }) => (
+        <Button
+          key={key}
+          type="button"
+          variant={variant}
+          size={size}
+          aria-label={`DTMF ${key}`}
+          {...buttonProps}
```
[the remark this answers is not in the document]
Comment (Regaddi) [reply package/src/components/elements/DTMFKeypad.tsx]: Fixed in 4332b10. `type="button"` is now applied after the `buttonProps` spread, so a consumer-supplied `buttonProps.type` can no longer turn keypad keys into form submitters. The same commit also composes `onClick`/`disabled` after the spread for the same reason.

File: package/src/components/elements/SessionInfo.tsx [hunk 3]
```diff
@@ -40,9 +40,11 @@ export const SessionInfo: React.FC<Props> = ({
...
     transportTypeName = `Daily (v${Daily.version()})`;
   } else if (
-    // @ts-expect-error - __proto__ not typed
-    client?.transport.__proto__.constructor.SERVICE_NAME ===
-    "small-webrtc-transport"
+    (
+      client?.transport as unknown as {
+        __proto__: { constructor: { SERVICE_NAME?: string } };
+      } | null
+    )?.__proto__.constructor.SERVICE_NAME === "small-webrtc-transport"
```
[the remark this answers is not in the document]
Comment (Regaddi) [reply package/src/components/elements/SessionInfo.tsx:47]: Leaving this one as-is for now. This line is part of an intentional workaround for a client-react published-types bug and is slated to be reverted wholesale once a fixed client-react release ships. The `__proto__` access predates this change (the branch only re-typed it), and the current ESLint config passes on it, so we would rather not churn the revert-later code here. Captured on the follow-up so it can be cleaned up (or dropped) alongside the workaround revert.

File: package/src/components/elements/DTMFKeypad.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+};
+
+Buffered.storyName = "Buffered (Pure)";
+
+/**
+ * Connected keypad. Requires a live Pipecat client; the keypad is disabled
+ * until the transport is connected. Pressing a key calls `sendDTMF`.
+ */
+export const Connected: Story<{
+  mode: "immediate" | "buffered";
```
Comment (markbackman) [inline package/src/components/elements/DTMFKeypad.stories.tsx:142]: What's the idea of the buffered mode? Pipecat already does aggregation on its side in the `DTMFAggregator`. I think we might want to keep it simple and just support immediate.

Comment (markbackman) [reply package/src/components/elements/DTMFKeypad.stories.tsx:142]: Ahh, I see. You have additional UI for buffered mode. Rather than have modes, should the user have access to both affordances all the time? That is, both a keypad and a text box?

**Review by markbackman — COMMENTED** [review]
The only other thought I have is whether we should play audio for keypresses. We have .wav files that you could use in the Pipecat repo:
https://github.com/pipecat-ai/pipecat/tree/main/src/pipecat/audio/dtmf

This may provide feedback to improve the UX. Otherwise, I noticed that with button presses, it's unclear whether my input was received. WDYT?

**Comment by markbackman** [comment]
Ok, I set up the example to run in buffered mode. Now I'm conflicted. Perhaps we just simplify the whole thing and _only_ have the buffered mode, no tones. The buffered mode itself solves the UX issue I had with the immediate mode.

Or, perhaps we keep both modes. Make buffered the default and add tones to the immediate mode to provide a better UX. WDYT? I think I'm leaning towards buffered only, but you may have thought about this more than me.

File: package/src/components/elements/DTMFKeypad.stories.tsx (hunk 4 above)
Comment (Regaddi) [reply package/src/components/elements/DTMFKeypad.stories.tsx:142]: The main idea here is to support both cases without headaches, so people wanting a keypad with immediate sending can have it, and people wanting control and the ability to review and edit can have that, too.
The sound feedback is a good note, I'll add that 👍

Comment (markbackman) [reply package/src/components/elements/DTMFKeypad.stories.tsx:142]: Yup! Makes sense. That's find of where I landed. The question is: which default to use for Pipecat Prebuilt 🤔

Comment (Regaddi) [reply package/src/components/elements/DTMFKeypad.stories.tsx:142]: Went with **buffered** as the default, for both the component and the Console's `keypadMode`.

The reasoning: immediate's per-keypress semantics really earn their keep when you're navigating someone else's phone tree, where each key gets its own response. But Prebuilt users are sending DTMF to their own Pipecat bot, which (as you pointed out) aggregates server-side anyway, so immediate buys little there. Buffered is also the more forgiving default, since a mistyped digit can be fixed with backspace before anything is transmitted, whereas an immediate send is irreversible.

Immediate is still a one-liner away via `mode="immediate"` / `keypadMode="immediate"` for the IVR case.

Worth noting on timing: the keypad is new in this PR, so flipping the default now is free. Changing it after release would be a behavior change for anyone already using the Console, which is why it seemed worth settling before merge.

**Comment by Regaddi** [comment]
Added tone feedback on keypresses, in both modes. Thanks for the nudge, the "did that even register?" problem was real, and it's what made buffered look better than it deserved credit for.

One deviation from your suggestion: instead of bundling the `.wav` files, the tones are synthesized. DTMF is literally dual-tone multi-frequency, so each key is just its two defining frequencies (`1` = 697 + 1209 Hz) played through a pair of WebAudio oscillators. That's a few lines and zero bytes of assets, which felt like a better fit for an npm UI kit than shipping 12 binaries and teaching tsup to inline them.

To be explicit about what this is: it's a local sidetone for feedback only. The actual signalling still goes over `sendDTMF`, so the tone is cosmetic and silencing it changes nothing about what the bot receives.

Two props to control it:
* `noToneFeedback` to turn it off. Named that rather than `noTones` so it can't be misread as "don't send DTMF".
* `toneVolume` (0 to 1, default 0.15), since the right level depends on how it sits against the call audio. Kept quiet by default because it plays over a live call.

Also flipped the default to buffered per the thread above, and tidied the branch down to 3 commits.

**Review by markbackman — APPROVED** [review]
Very nice!!
